### PR TITLE
Search: Minor improvements to front end filtering UI

### PR DESCRIPTION
**Pull request by gibrown**
- Show "remove all" if any active filters. This feels more consistent to me.
- Add arrow to "remove all filters". Just a link looked kinda lonely.
- Remove bullets in lists. eek bullets and checkboxes. :)

Here's what it now looks like (I forgot to get a before):

screen shot 2018-01-26 at 10 07 10 am

### CODE REVIEW

**Comment by ebinnion** [comment]
Code-wise, this LGTM. @MichaelArestad has some design feedback in Slack.

**Comment by Viper007Bond** [comment]
> I forgot to get a before

You can just switch back to `master` and screenshot. ;)

**Comment by ebinnion** [comment]
I'm not sure if you're ready for a review. But, I wanted to be sure to point the bug out.

**Comment by gibrown** [comment]
Nice catch @ebinnion I think it should be fixed now.

**Comment by ebinnion** [comment]
It looks like the issue is fixed when removing post types from the post type filter. 

But, if I select both a post type filter and a month filter for example:

screen shot 2018-01-26 at 5 13 55 pm

Then the "Clear Filters" link for the month filter looks like this:

`?s=content&orderby=relevance&order=DESC&post_type=post,page,jetpack-testimonial`

In this case, I would expect `post_type=post` since I have the `post` filter selected. I think we only have to modify the "Clear Filters" link to have the post types if it's a `post_type` filter.

**Comment by gibrown** [comment]
Finally finished it off i think.

screen shot 2018-01-29 at 3 04 40 pm

File: modules/search/class.jetpack-search-helpers.php [hunk 1]
```diff
@@ -349,6 +349,24 @@ static function get_filter_properties_for_tracks( $filters ) {
 		return $filters_properties;
 	}
 
+	/**
+	 * Gets the active post types given a set of filters.
+	 *
+	 * @param array $filters The active filters for the current query.
+	 * @param array $default_post_types The default post types.
+	 *
+	 * @return boolean
```
Comment (ebinnion) [inline modules/search/class.jetpack-search-helpers.php]: I think this should be `@return array`

**Comment by MichaelArestad** [comment]
I think we can merge this as is and iterate. I think the next iteration will involve more opinionated style changes.

File: modules/search/class.jetpack-search-widget-filters.php [hunk 2]
```diff
@@ -736,16 +729,37 @@ function render_widget_title( $title, $before_title, $after_title ) {
...
+				$clear_url = Jetpack_Search_Helpers::add_post_types_to_url( $clear_url, $default_post_types );
+			}
+		}
+
 		?>
 		<h4 class="jetpack-search-filters-widget__sub-heading">
 			<?php echo esc_html( $filter['name'] ); ?>
 		</h4>
+		<?php if ( $clear_url ) : ?>
+		<div class="jetpack-search-filters-widget__clear"><a href="<?php echo esc_url( $clear_url ); ?>">
```
Comment (ebinnion) [inline modules/search/class.jetpack-search-widget-filters.php]: This is minor, but the indentation is off here. It should probably be:

```
<div>
    <a>
        echo
    </a>
</div>
```

**Comment by gibrown** [comment]
@gravityrail @ebinnion ya, can someone approve this and let's merge it, we've been back and forth on language and icons a bunch of times.

**Pull request merged**